GeneralsMD/Code/GameEngine/Include/GameLogic/Object.h

```diff
@@ -440,6 +440,8 @@ class Object : public Thing, public Snapshot
 	// contained-by
 	inline Object *getContainedBy() { return m_containedBy; }
 	inline const Object *getContainedBy() const { return m_containedBy; }
+	Object *getContainedBySpecialOverlordStyle(); ///< Get parent container object if its contain module is Overlord style.
+	const Object *getContainedBySpecialOverlordStyle() const; ///< Get parent container object if its contain module is Overlord style.
```

Comment:
Is the `Object` class the best place for these functions? It feels backwards to put specific behaviour here (despite most of the code base already breaking this principle).

Reply:
Do you have suggestion where to put it?

Reply:
Ideally the container should be aware of whether its occupants are exposed / visible (I believe `isEnclosingContainerFor` might typically be used for this purpose). This would then allow all non-enclosed contained objects to provide their Propaganda effects, such as infantry in a Firebase if the `PropagandaTowerBehavior` module were to be added to them.

Reply:
Creating endless fix proposals is inefficient over simply discussing the solution. Which parts are unclear?

If #1545 is not generic enough then I suggest this change is not either. The ideal solution also supports infantry with `PropagandaTowerBehavior` modules in Firebases.

Reply:
There is a field on the `GarrisonContain` module - `IsEnclosingContainer` that is set to false for the Firebase (and dynamically returned for `OverlordContain::isEnclosingContainerFor`). Perhaps similar logic is needed in this case?